### PR TITLE
Create Mapping Target FBs if not Present

### PR DESCRIPTION
**Pull request by azoitl**
Based on the mapping source the target FB is created inside of the resource. This is the starting point of not requiring to store mapped information as part of the resource.

### CODE REVIEW

File: plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/dataimport/MappingTargetCreator.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+		};
+	}
+
+	private static void setupConfigureableFB(final ConfigurableFB srcConfFB, final ConfigurableFB targetConfFB) {
+		targetConfFB.setDataType(srcConfFB.getDataType());
+		targetConfFB.updateConfiguration();
+	}
+
+	private static String extractTargetFBName(final String targetFBName) {
+		String name = targetFBName;
```
Comment (oberlehner) [inline plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/dataimport/MappingTargetCreator.java]: do you need this local variable ?

File: plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/dataimport/MappingTargetCreator.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+	public static void transferFBParams(@NonNull final FBNetworkElement srcElement,
+			@NonNull final FBNetworkElement targetElement) {
+		final List<VarDeclaration> destInputs = targetElement.getInterface().getInputVars();
+		final List<VarDeclaration> srcInputs = srcElement.getInterface().getInputVars();
+
+		for (int i = 0; i < destInputs.size(); i++) {
+			final VarDeclaration srcVar = srcInputs.get(i);
+			final VarDeclaration dstVar = destInputs.get(i);
+
+			if ((null != srcVar.getValue()) && (!srcVar.getValue().getValue().isEmpty())) {
```
Comment (oberlehner) [inline plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/dataimport/MappingTargetCreator.java]: Please be consistent with you null checks.
in line 53 you write srcElement.getTypeEntry() != null and here youwrite null != ....

Comment (oberlehner) [reply plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/dataimport/MappingTargetCreator.java]: Can't you use the change value command here ?

Comment (azoitl) [reply plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/dataimport/MappingTargetCreator.java]: No I can not use any commands in the model plugin.

Comment (azoitl) [reply plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/dataimport/MappingTargetCreator.java]: fixed

File: plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/dataimport/MappingTargetCreator.java (hunk 1 above)
Comment (azoitl) [reply plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/dataimport/MappingTargetCreator.java]: no, thx for noting. fixed.

**Review by oberlehner — APPROVED** [review]

File: plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/dataimport/MappingTargetCreator.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Alois ZOitl - initial implementation
+ *******************************************************************************/
```
Comment (mx990) [inline plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/dataimport/MappingTargetCreator.java]: Please remove the duplicate copyright header.

File: plugins/org.eclipse.fordiac.ide.model.commands/src/org/eclipse/fordiac/ide/model/commands/change/MapToCommand.java [hunk 4]
```diff
@@ -157,21 +159,16 @@ public void redo() {
...
+		case final UntypedSubApp untypedSubapp -> createTargetUntypedSubApp();
+		default -> null;
+		};
 		if (created != null) {
 			created.setName(srcElement.getName());
-			transferFBParams(srcElement, created);
+			MappingTargetCreator.transferFBParams(srcElement, created);
 		}
 		return created;
 	}
```
Comment (mx990) [inline plugins/org.eclipse.fordiac.ide.model.commands/src/org/eclipse/fordiac/ide/model/commands/change/MapToCommand.java:174]: Could we also replace this with `MappingTargetCreator.createMappingTarget(...)`?

File: plugins/org.eclipse.fordiac.ide.model.commands/src/org/eclipse/fordiac/ide/model/commands/change/MapCommunicationCommand.java [hunk 5]
```diff
@@ -44,7 +45,7 @@ protected FBNetworkElement createTargetElement() {
 		comm.setPosition(EcoreUtil.copy(srcElement.getPosition()));
 		comm.setTypeEntry(srcElement.getTypeEntry());
 		comm.setInterface(srcElement.getType().getInterfaceList().copy());
-		transferFBParams(srcElement, comm);
+		MappingTargetCreator.transferFBParams(srcElement, comm);
```
Comment (mx990) [inline plugins/org.eclipse.fordiac.ide.model.commands/src/org/eclipse/fordiac/ide/model/commands/change/MapCommunicationCommand.java:48]: If we copy the interface directly from `srcElement`, could we avoid transferring the FB params?

**Review by mx990 — CHANGES_REQUESTED** [review]

File: plugins/org.eclipse.fordiac.ide.model.commands/src/org/eclipse/fordiac/ide/model/commands/change/MapToCommand.java (hunk 4 above)
Comment (azoitl) [reply plugins/org.eclipse.fordiac.ide.model.commands/src/org/eclipse/fordiac/ide/model/commands/change/MapToCommand.java:174]: I have planned that for a later commit yes.

File: plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/dataimport/MappingTargetCreator.java (hunk 3 above)
Comment (azoitl) [reply plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/dataimport/MappingTargetCreator.java]: ah damn how did that happen.

File: plugins/org.eclipse.fordiac.ide.model.commands/src/org/eclipse/fordiac/ide/model/commands/change/MapCommunicationCommand.java (hunk 5 above)
Comment (azoitl) [reply plugins/org.eclipse.fordiac.ide.model.commands/src/org/eclipse/fordiac/ide/model/commands/change/MapCommunicationCommand.java:48]: From what I tested in the MappingTargetCreator no.

**Review by mx990 — APPROVED** [review]

**Pull request merged**